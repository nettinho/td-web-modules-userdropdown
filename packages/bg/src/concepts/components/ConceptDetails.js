import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Segment, Header } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { RichTextEditor } from "@truedat/core/components";
import { DynamicFormViewer } from "@truedat/df/components";
import { getFieldValues } from "../selectors";

export const ConceptDetails = ({ concept, fieldValues }) => (
  <React.Fragment>
    <Segment attached="bottom">
      <Header as="h3">
        <FormattedMessage id="concepts.props.description" />
      </Header>
      <RichTextEditor readOnly value={concept.description} />
      <DynamicFormViewer fieldValues={fieldValues} />
    </Segment>
  </React.Fragment>
);

ConceptDetails.propTypes = {
  concept: PropTypes.object,
  fieldValues: PropTypes.array
};

const mapStateToProps = state => ({
  concept: state.concept,
  fieldValues: getFieldValues(state)
});

export default connect(mapStateToProps)(ConceptDetails);
