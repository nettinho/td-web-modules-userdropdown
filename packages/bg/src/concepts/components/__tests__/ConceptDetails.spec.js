import React from "react";
import { shallow } from "enzyme";
import { ConceptDetails } from "../ConceptDetails";

describe("<ConceptDetails />", () => {
  const props = {
    concept: { name: "Concept1", type: "type", status: "draft" },
    fieldValues: [
      ["group1", [{ field: "Field1", value: "Value1" }]],
      [
        "group2",
        [
          { field: "Field2", value: "Value2" },
          { field: "Field3", value: "Value3" }
        ]
      ]
    ]
  };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<ConceptDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
