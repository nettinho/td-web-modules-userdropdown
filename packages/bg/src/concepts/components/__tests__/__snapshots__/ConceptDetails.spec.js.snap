// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConceptDetails /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Segment
    attached="bottom"
  >
    <Header
      as="h3"
    >
      <FormattedMessage
        id="concepts.props.description"
        values={Object {}}
      />
    </Header>
    <RichTextEditor
      readOnly={true}
    />
    <Connect(DynamicFormViewer)
      fieldValues={
        Array [
          Array [
            "group1",
            Array [
              Object {
                "field": "Field1",
                "value": "Value1",
              },
            ],
          ],
          Array [
            "group2",
            Array [
              Object {
                "field": "Field2",
                "value": "Value2",
              },
              Object {
                "field": "Field3",
                "value": "Value3",
              },
            ],
          ],
        ]
      }
    />
  </Segment>
</React.Fragment>
`;
